.github/workflows/main.yml

```
@@ -0,0 +1,61 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  format:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Install OTP and Elixir
+        uses: ericmj/setup-elixir@2429dfc
+        with:
+          erlang-tag: OTP-22.1
+          elixir-tag: v1.9.4
+
+      - name: Install dependencies
+        run: mix deps.get
+
+      - name: Check mix format
+        run: mix format --check-formatted
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        pair:
+          - erlang: OTP-22.1
+            elixir: v1.9.4
+          - erlang: OTP-21.3
+            elixir: v1.8.2
+          - erlang: OTP-20.3.1
+            elixir: v1.7.4
+          - erlang: OTP-19.3
+            elixir: v1.6.6
+          - erlang: OTP-18.3
+            elixir: v1.5.3
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Install OTP and Elixir
+        uses: ericmj/setup-elixir@2429dfc
+        with:
+          erlang-tag: ${{matrix.pair.erlang}}
+          elixir-tag: ${{matrix.pair.elixir}}
+
+      - name: Install dependencies
+        run: |
+          mix deps.get
+          mix deps.compile
+
+      - name: Start docker
+        run: |
+          docker-compose build
+          docker-compose up --detach
```

```suggestion
          docker-compose up --build --detach
```

and you can remove the line above :)

---

I think in this case the `--build` flag is not necessary because `docker-compose up` builds the images if they don't exist 🙂  The `--build` flag means "force build".